src/app/molecules/message/Message.jsx

```diff
@@ -516,7 +526,7 @@ const MessageOptions = React.memo(({
             </MenuItem>
             <MenuItem
               iconSrc={CmdIC}
-              onClick={() => openViewSource(mEvent)}
+              onClick={() => openViewSource(editedMEvent !== undefined ? editedMEvent : mEvent)}
```

Comment:
creating `handleOpenViewSource` function and doing all editedMEvent finding stuff in that func and opening view source from there will be a good idea cuz then we are doing calc when we actually need them

Reply:
Does react do this calc `editedMEvent !== undefined ? editedMEvent : mEvent` for each rerender? I didn't quite get why you are suggesting us to create a new function `handleOpenViewSource`.

Would be cleaner. But my question is on if we are doing unnecessary calc

Reply:
I mean execution from line 484 to 490 on only require when user click menu item.

so we can do that in `handleOpenViewSource` and call `openViewSource` from there